src/MongoDB/WriteResult.c

```diff
@@ -97,17 +102,16 @@ PHP_METHOD(WriteResult, getModifiedCount)
 		return;
 	}
 
-	if (intern->write_result.omit_nModified) {
-		RETURN_NULL();
+	if (bson_iter_init_find(&iter, intern->reply, "nModified") && BSON_ITER_HOLDS_INT32(&iter)) {
+		RETURN_LONG(bson_iter_int32(&iter));
```

Comment:
libmongoc decides to include an `nModified` field in the `bson_t` based on `omit_nModified`. See: https://github.com/mongodb/mongo-c-driver/blob/d538a9c6fcd44b4fdf8e112b9223362a32e6e168/src/mongoc/mongoc-write-command.c#L1930